idaes/models_extra/gas_solid_contactors/unit_models/tests/test_FB1D.py

```diff
@@ -0,0 +1,1367 @@
+                * b.gas_phase.properties[tf, x].enth_mol
+                * b.gas_phase.area[tf, x]
+            )
+            energy_initial_per_length = (
+                b.gas_phase.properties[t0, x].dens_mol
+                * b.gas_phase.properties[t0, x].enth_mol
+                * b.gas_phase.area[t0, x]
+            )
+            return (energy_final_per_length - energy_initial_per_length) * b.bed_height
+
+        iron_oc.fs.unit.gas_energy_holdup = Integral(
+            iron_oc.fs.unit.length_domain,
+            wrt=iron_oc.fs.unit.length_domain,
+            rule=gas_energy_holdup,
+            doc="Gas energy holdup, J",
+        )
+
+        def oc_energy_at_initial_time(b, x):
+            t0 = b.flowsheet().time.first()
+            energy_per_length = (
+                b.solid_properties[t0, x].dens_mass_particle
+                * b.solid_properties[t0, x].enth_mass
+                * b.solid_phase_area[t0, x]
+            )
+            return energy_per_length * b.bed_height
+
+        iron_oc.fs.unit.oc_energy_at_initial_time = Integral(
+            iron_oc.fs.unit.length_domain,
+            wrt=iron_oc.fs.unit.length_domain,
+            rule=oc_energy_at_initial_time,
+            doc="OC energy at initial reaction time, J",
+        )
+
+        def oc_energy_at_final_time(b, x):
+            tf = b.flowsheet().time.last()
+            energy_per_length = (
+                b.solid_properties[tf, x].dens_mass_particle
+                * b.solid_properties[tf, x].enth_mass
+                * b.solid_phase_area[tf, x]
+            )
+            return energy_per_length * b.bed_height
+
+        iron_oc.fs.unit.oc_energy_at_final_time = Integral(
+            iron_oc.fs.unit.length_domain,
+            wrt=iron_oc.fs.unit.length_domain,
+            rule=oc_energy_at_final_time,
+            doc="OC at final reaction time, J",
+        )
+
+        if iron_oc.fs.unit.config.solid_phase_config.reaction_package is not None:
+            e_reaction = (
+                value(
+                    (
+                        iron_oc.fs.unit.fe2o3_at_final_time_mol
+                        - iron_oc.fs.unit.fe2o3_at_initial_time_mol
+                    )
+                    * iron_oc.fs.unit.solid_reactions[0, 1]._params.dh_rxn["R1"]
+                )
+                / 12
+            )
+        else:
+            e_reaction = 0
+
+        enthalpy_change_gas = value(  # in - out - holdup/accumulation
+            iron_oc.fs.unit.gas_input_energy_total
+            - iron_oc.fs.unit.gas_output_energy_total
+            - iron_oc.fs.unit.gas_energy_holdup
+        )
+
+        enthalpy_change_solid = value(
+            iron_oc.fs.unit.oc_energy_at_final_time
+            - iron_oc.fs.unit.oc_energy_at_initial_time
+            - e_reaction
+        )
+
+        ebal_abs_tol = enthalpy_change_solid - enthalpy_change_gas
+        ebal_rel_tol = ebal_abs_tol / enthalpy_change_solid
+        assert ebal_rel_tol <= 0.15
```

Comment:
And here as well for the balance checks.

Reply:
Fixed.